api/librdb-api.h

```diff
@@ -159,75 +166,144 @@ typedef struct RdbStreamConsumerMeta {
+    /* Callback to handle a ziplist-based list value */
+    RdbRes (*handleListZL)(RdbParser *p, void *userData, RdbBulk ziplist);
+    /* Callback to handle a listpack-based list value */
+    RdbRes (*handleListLP)(RdbParser *p, void *userData, RdbBulk listpack);
+
+    /* Callback to handle a field-value pair within a plain-hash */
+    RdbRes (*handleHashPlain)(RdbParser *p, void *userData, RdbBulk field, RdbBulk value);
+    /* Callback to handle a ziplist-based hash value */
+    RdbRes (*handleHashZL)(RdbParser *p, void *userData, RdbBulk ziplist);
+    /* Callback to handle a listpack-based hash value */
+    RdbRes (*handleHashLP)(RdbParser *p, void *userData, RdbBulk listpack);
+    /* Callback to handle a zipmap-based hash value */
+    RdbRes (*handleHashZM)(RdbParser *p, void *userData, RdbBulk zipmap);
+
+    /* Callback to handle an item from a plain-hash */
+    RdbRes (*handleSetPlain)(RdbParser *p, void *userData, RdbBulk item);
+    /* Callback to handle an intset-based set value */
+    RdbRes (*handleSetIS)(RdbParser *p, void *userData, RdbBulk intset);
+    /* Callback to handle a listpack-based set value */
+    RdbRes (*handleSetLP)(RdbParser *p, void *userData, RdbBulk listpack);
+
+    /*** TODO: RdbHandlersStructCallbacks: ***/
+
+    /* Callback to handle a ziplist-based sorted set value */
+    RdbRes (*handleZsetZL)(RdbParser *p, void *userData, RdbBulk ziplist);
+    /* Callback to handle a listpack-based sorted set value */
+    RdbRes (*handleZsetLP)(RdbParser *p, void *userData, RdbBulk listpack);
+    /* Callback to handle function code */
     RdbRes (*handleFunction)(RdbParser *p, void *userData, RdbBulk func);
+
     /*** TODO: RdbHandlersStructCallbacks: stream stuff ... ***/
+
+    /* Callback to handle a stream key with listpack value */
     RdbRes (*handleStreamLP)(RdbParser *p, void *userData, RdbBulk nodekey, RdbBulk streamLP);
 
 } RdbHandlersStructCallbacks;
 
+/****************************************************************
+ * Redis Data Type Callbacks for Registration (handlers Level 2)
+ *
+ * This level of callback registration is adequate when the focus is solely on
+ * the logical data types within the database, such as to export the data to
+ * another framework or investigate the content of the data.
+ ****************************************************************/
+
 typedef struct RdbHandlersDataCallbacks {
     HANDLERS_COMMON_CALLBACKS
+    /* Callback to handle the value of a string key */
     RdbRes (*handleStringValue)(RdbParser *p, void *userData, RdbBulk str);
-    RdbRes (*handleListElement)(RdbParser *p, void *userData, RdbBulk str);
+    /* Callback to handle an item from a list */
+    RdbRes (*handleListItem)(RdbParser *p, void *userData, RdbBulk item);
+    /* Callback to handle a field-value pair within a hash */
+    RdbRes (*handleHashFieldValue)(RdbParser *p, void *userData, RdbBulk field, RdbBulk value);
```

Comment:
i think we can drop the "value" part.
we don't use it in ZSET (although we do have a score)/